go/processor/floria/calls.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package floria
+
+import (
+	"fmt"
+
+	"github.com/Fantom-foundation/Tosca/go/tosca"
+)
+
+func Call(interpreter tosca.Interpreter, transaction tosca.Transaction, context tosca.TransactionContext) (tosca.Result, error) {
+
+	blockParameters := tosca.BlockParameters{}
+
+	transactionParameters := tosca.TransactionParameters{
+		Origin:     transaction.Sender,
+		GasPrice:   transaction.GasPrice,
+		BlobHashes: []tosca.Hash{},
+	}
+
+	codeHash := context.GetCodeHash(*transaction.Recipient)
+	code := context.GetCode(*transaction.Recipient)
+
+	params := tosca.Parameters{
+		BlockParameters:       blockParameters,
+		TransactionParameters: transactionParameters,
+		//Context:               runContext, todo implement
+		Kind:      tosca.Call,
+		Static:    false,
+		Depth:     0, // todo add depth check
+		Gas:       transaction.GasLimit,
+		Recipient: *transaction.Recipient,
+		Sender:    transaction.Sender,
+		Input:     transaction.Input,
+		Value:     transaction.Value,
+		CodeHash:  &codeHash,
+		Code:      code,
+	}
+
+	snapshot := context.CreateSnapshot()
+	if err := transferValue(transaction, context); err != nil {
+		context.RestoreSnapshot(snapshot)
+		return tosca.Result{}, err
```

Comment:
Something to be covered in a follow-up: we need to check whether in this case the full Gas price should be charged.

Reply:
In [geth](https://github.com/ethereum/go-ethereum/blob/cf0378499f1bcae65c093c58cd6ca8225e91b125/core/vm/evm.go#L204), the unchanged gas in returned if there is not enough balance. Checks for this will be added once the gas computation and billing is implemented.